### ISSUE TITLE
[Chapter 5.3] Continue hangs, stuck at "0x00000000 in __vector_table ()"

### ISSUE THREAD

**Issue by garrelj1** [rust-embedded/discovery#564]
Running on Kali Linux

I'm able to build and flash the code. Once I try to debug, I get the following:

```
(gdb) target remote :1337
Remote debugging using :1337
warning: while parsing target description (at line 4): Target description specified unknown architecture "armv7"
warning: Could not load XML target description; ignoring
0x00000000 in __vector_table ()
```

Notice `0x00000000 in __vector_table ()`

After adding a break point, and continuing, it just hangs:

```
(gdb) b main
Breakpoint 1 at 0x15c: file src/05-led-roulette/src/main.rs, line 9.
Note: automatically using hardware breakpoints for read-only addresses.

(gdb) continue
Continuing.

```

I've also tried `step`, which gives

```
(gdb) step
Single stepping until exit from function __vector_table,
which has no line number information.

Program received signal SIGINT, Interrupt.
0x00000000 in __vector_table ()
```

And I've tried `stepi`, which gives

```
(gdb) stepi

Program received signal SIGINT, Interrupt.
0x00000000 in __vector_table ()
```

Any ideas of how else to debug or get further?

**Comment by Michal2404** [comment]
Hey, Did you manage to resolve this issue? I am currently hanging on it and have no idea what is the cause...

**Comment by BartMassey** [comment]
Could you please take a look at https://docs.rust-embedded.org/discovery-mb2 and https://github.com/rust-embedded/discovery-mb2 and see if you can reproduce this issue there? If so, please file an issue there and we'll try to get it figured out and fixed.

**Comment by Michal2404** [comment]
Have you tried running **gdb-multiarch** instead of **gdb**? I had the same issue and it helped me.

**Comment by zanebliss** [comment]
@Michal2404 What did you do exactly to fix this? Did you use `gbd-multiarch` as the command to run instead of `gdb`?

**Comment by Michal2404** [comment]
Exactly. It depends on which version of gdb you have installed (chapter 3 of the book).

**Comment by garrelj1** [comment]
I haven't tried myself (been working other stuff), but others seem to have gotten this working using `gdb-multiarch`. I'll close the issue
